lib/ecto/query/builder.ex

```diff
@@ -178,6 +180,26 @@ defmodule Ecto.Query.Builder do
      {params |> wrap_nil(left) |> wrap_nil(right), acc}}
   end
 
+  # mathematical operators
+  def escape({math_op, _, [left, right]} = expr, type, params_acc, vars, env)
+      when math_op in ~w(+ - * /)a do
+    assert_type!(expr, type, :number)
+    
+    args_type =
```

Comment:
I think `args_type` should always be `:number` and then we handle the case of `:number` later on quoted_type implementation.